packages/jaeger-ui/src/model/link-patterns.tsx

```diff
@@ -119,20 +140,17 @@ function callTemplate(template: ProcessedTemplate, data: any) {
 
 export function computeTraceLink(linkPatterns: ProcessedLinkPattern[], trace: Trace) {
   const result: TLinksRV = [];
-  const validKeys = (Object.keys(trace) as (keyof Trace)[]).filter(
-    key => typeof trace[key] === 'string' || typeof trace[key] === 'number'
-  );
 
   linkPatterns
     .filter(pattern => pattern.type('traces'))
     .forEach(pattern => {
       const parameterValues: Record<string, any> = {};
       const allParameters = pattern.parameters.every(parameter => {
-        const key = parameter as keyof Trace;
-        if (validKeys.includes(key)) {
+        const val = getParameterInTrace(parameter, trace);
```

Comment:
it looks like `validKeys` will now be re-calculated for every pattern and every parameter. Perhaps this needs a helper class:
```
const traceParams = TraceParams(trace);
// in a loop
  const val = traceParams.getParameter(parameter);
```

Reply:
There is a helper function `getValidTraceKeys()` which solves memoize the returned value for a trace object, let me know if that is not ok.

Reply:
ah, indeed, I forgot the memoize - probably ok then.